### DISCUSSION TITLE
Discussion on Reasons behind ResNet Modification

### DISCUSSION THREAD

**Discussion by LiyaoTang** [STVIR/pysot#191] [General]
Thank you guys for such a useful opensource VOT repo. it really helps us understand a lot more on tracking.

Yet, while I am learning the code. I did not understand a few modifications in our ResNet backbone, while I do understand those changes are made to adapt for 255/127 input size.

More specifically, I do not understand the logic behind the following two parts:

1. the padding-dilation logic in BasicBlock and Bottleneck:
![Screenshot from 2019-09-28 17-24-15](https://user-images.githubusercontent.com/17057383/65814495-31405100-e215-11e9-84c2-009cbcf96c33.png)

2. the logic to modify dilation & padding in make_layers(), and it seems the highlighted logic "if" block is used for layer-3 and 4, and "else" block is used for layer-2.
![Screenshot from 2019-09-28 17-24-31](https://user-images.githubusercontent.com/17057383/65814556-ed9a1700-e215-11e9-9060-9b02eb3aee54.png)

The reason why I am interested in this is that I would like to know if the structure can be extended to build ResNet-101 for siamTracker? Or, it is doing logic trick that MAY only work for resnet-50?

Thank you in advance. ;)

**Comment by zhengjyu** [comment]
Excause me, do you know the reason of the dilation // 2 ?
